igcollect/ping.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+"""igcollect - Latency collection
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+# This script is used to collect ping data to evaluate availability and latency
+# of our network and make it easy to create a Graphite Dashboard
+#
+# Please note that the package fping has to be installed on your Linux server!
+
+
+import argparse
+import subprocess
+import time
+
+from multiprocessing import Pool
+from os import cpu_count
+from statistics import stdev
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description='Ping collector script for Graphite ')
+    parser.add_argument('hosts', nargs='+',
+                        help='the hosts to ping')
+    parser.add_argument('-c', '--count', type=int, default=20,
+                        choices=range(1,60),
+                        help='how many times the script will try to ping '
+                             '(number has to be bigger '
+                             'than 1 and smaller than 60)')
+    parser.add_argument('-p', '--prefix',
+                        default='ping',
+                        help='the path to the value in Graphite ')
+    parser.add_argument('-t', '--timeout', type=int, default=500,
+                        help='Initial target timeout in milliseconds. '
+                             'the amount of time that program '
+                             'waits for a response')
+    parser.add_argument('-d', '--delay', type=int, default=1800,
+                        choices=range(0,5000),
+                        help='time to wait between a series of pings in '
+                             'milliseconds')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    chunked_hosts = [args.hosts[i:i + 10]
+                     for i in range(0, len(args.hosts), 10)]
+
+    data = [(args.prefix, chunk, args.count, args.timeout, args.delay)
+            for chunk in chunked_hosts]
+
+    with Pool(cpu_count() * 2) as p:
+        p.starmap(check_pings, data)
+
+
+def check_pings(prefix, hosts, count, timeout, delay):
```

Comment:
- [ ] Do not forget the doc strings for the functions

Reply:
As for utility functions it is absolutely meaningful to have, I can't really see the benefit for the `main` and `parse_args` function. The `main` function should be explain in the script description already and the `parse_args` function partly already in the script explanation as for how to use the script and also mostly redundant if the `parse_args` is not complex.
What is your opinion about that @kofrezo?

Reply:
Agreed. I would not complain if there is no doc string for the main and parse_args function but everything else should have one.